File: src/main/java/fr/univ_smb/isc/m1/brutocollinus/model/arsenal/ArmedBruto.java
package fr.univ_smb.isc.m1.brutocollinus.model.arsenal;

public class ArmedBruto {
    private String name;
    private IBrutoClass brutoClass;

    public ArmedBruto(String name, IBrutoClass brutoClass) {
        this.name = name;
    }

    public String name() {
        return this.name;
    }

    public FightStatistics fightStatistics() {
        return this.brutoClass.fightStatistics();
    }
}

File: frontend/src/components/ArmedBruto.js
import React from 'react'
import {ListGroup} from 'react-bootstrap';
import FightStatistics from './FightStatistics.js';
import Boost from './Boost.js';
import Stuff from './Stuff.js';
import BrutoClass from './BrutoClass.js';

export default function ArmedBruto(props) {
    const armedBruto = props.armedBruto;

    return (
        <div className={props.className}>
            <div className="text-center">
                <h4>{ armedBruto.name }</h4>
                <FightStatistics fightStatistics={armedBruto.totalFightStatistics} />
            </div>

            <h6>Classe</h6>
            <BrutoClass brutoClass={armedBruto.brutoClass}/>

            <h6>Boosts</h6>
            <ListGroup>
                {armedBruto.equipedBoosts.map(boost =>
                    (<ListGroup.Item>
                        <Boost boost={boost}/>
                    </ListGroup.Item>))}
            </ListGroup>

            <h6>Stuffs</h6>
            <ListGroup>
                {armedBruto.equipedStuffs.map(stuff =>
                    (<ListGroup.Item>
                        <Stuff stuff={stuff}/>
                    </ListGroup.Item>))}
            </ListGroup>
        </div>
    );
}